initrd/bin/oem-factory-reset

```diff
@@ -60,195 +105,206 @@ SKIP_BOOT="n"
 
 die() {
 
-    local msg=$1
-    if [ -n "$msg" ]; then
-        echo -e "\n$msg"
-    fi
-    kill -s TERM $TOP_PID
-    exit 1
+	local msg=$1
+	if [ -n "$msg" ]; then
+		echo -e "\n$msg"
+	fi
+	kill -s TERM $TOP_PID
+	exit 1
 }
 
 local_whiptail_error() {
-    local msg=$1
-    if [ "$msg" = "" ]; then
-        die "whiptail error: An error msg is required"
-    fi
-    whiptail_error --msgbox "${msg}\n\n" $HEIGHT $WIDTH --title "Error"
+	local msg=$1
+	if [ "$msg" = "" ]; then
+		die "whiptail error: An error msg is required"
+	fi
+	whiptail_error --msgbox "${msg}\n\n" $HEIGHT $WIDTH --title "Error"
 }
 
 whiptail_error_die() {
-    local_whiptail_error "$@"
-    die
+	local_whiptail_error "$@"
+	die
 }
 
 mount_boot() {
-    TRACE_FUNC
-    # Mount local disk if it is not already mounted.
-    # Added so that 'o' can be typed early at boot to enter directly into OEM Factory Reset
-    if ! grep -q /boot /proc/mounts; then
-        # try to mount if CONFIG_BOOT_DEV exists
-        if [ -e "$CONFIG_BOOT_DEV" ]; then
-            mount -o ro $CONFIG_BOOT_DEV /boot || die "Failed to mount $CONFIG_BOOT_DEV. Please change boot device under Configuration > Boot Device"
-        fi
-    fi
+	TRACE_FUNC
+	# Mount local disk if it is not already mounted.
+	# Added so that 'o' can be typed early at boot to enter directly into OEM Factory Reset
+	if ! grep -q /boot /proc/mounts; then
+		# try to mount if CONFIG_BOOT_DEV exists
+		if [ -e "$CONFIG_BOOT_DEV" ]; then
+			mount -o ro $CONFIG_BOOT_DEV /boot || die "Failed to mount $CONFIG_BOOT_DEV. Please change boot device under Configuration > Boot Device"
+		fi
+	fi
+}
+
+reset_nk3_secret_app() {
+	TRACE_FUNC
+	# Reset Nitrokey 3 secret app
+	if lsusb | grep -q "20a0:42b2"; then
+		echo
+		echo "Resetting Nitrokey 3 secret app"
+		# Reset Nitrokey 3 secret app
+		/bin/hotp_verification reset
```

Comment:
a SECRET_APP_PIN is definitely needed here, no previous PIN default set: we need to create one, ideally with same value as GPG Admin PN to not confuse UX even more.